client/src/components/EditDetails.jsx

```diff
@@ -1,60 +1,68 @@
+import React from 'react'
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem,
+} from 'reactstrap'
+import '!style-loader!css-loader!bootstrap/dist/css/bootstrap.css'
+import axios from 'axios'
+
+const Months = Array.from(Array(12)).map(i => i + 1)
+const Days = Array.from(Array(31)).map(i => i + 1)
 
 class EditDetails extends React.Component {
   constructor(props) {
-    super(props);
-    let { height, weight, age, gender, dob, zip } = props.profile;
+    super(props)
+    let { height, weight, age, gender, dob, zip } = props.profile
+
+    const [heightFt, heightIn] = height.split('/')
+
     this.state = {
-      heightFt: height.split('/')[0],
-      heightIn: height.split('/')[1],
-      weight: weight.split(' ')[0],
       age,
-      gender,
       dob,
+      gender,
+      heightFt,
+      heightIn,
       zip,
       dropdownOpen: false,
-    };
-    this.saveChanges = this.saveChanges.bind(this);
-    this.toggle = this.toggle.bind(this);
-    this.clickDropdown = this.clickDropdown.bind(this);
-    this.changeHandler = this.changeHandler.bind(this);
-  };
-
-  changeHandler(e) {
-    this.setState({ [e.target.name]: e.target.value }, () => console.log(this.state));
+      weight: weight.split(' ')[0],
+    }
   }
 
-  toggle() {
+  changeHandler = e => {
+    this.setState({ [e.target.name]: e.target.value }, () =>
+      console.log(this.state),
+    )
+  }
+
+  toggle = () => {
     this.setState({
-      dropdownOpen: !this.state.dropdownOpen
-    });
-  };
+      dropdownOpen: !this.state.dropdownOpen,
+    })
+  }
 
-  clickDropdown(e) {
-    this.setState({ gender: e.target.innerText });
+  clickDropdown = e => {
+    this.setState({ gender: e.target.innerText })
   }
 
-  saveChanges(e) {
-    let { heightFt, heightIn, weight, age, dob, zip } = this.state;
+  saveChanges = e => {
+    let { heightFt, heightIn, weight, age, dob, zip } = this.state
+
     console.log('typeof:', Number(heightIn))
-    let monthArr = [ ...Array(12).keys() ].map( i => i+1);
-    let dayArr = [ ...Array(31).keys() ].map( i => i+1);
-    if (Number(heightFt) === NaN || Number(heightIn) === NaN || Number(weight) === NaN || Number(age) === NaN || Number(zip) === NaN
-    || !monthArr.includes(Number(dob.split('/')[0])) || !dayArr.includes(Number(dob.split('/')[1])) || dob.split('/')[2].length !== 4 
-    || Number(dob.split('/')[2]) === NaN) {
-      alert('please enter valid input');
-    } else {
-      window.location.reload();
-      let details = this.state;
-      details.height = `${this.state.heightFt}/${this.state.heightIn}`;
-      delete details.heightFt;
-      delete details.heightIn;
-      details.weight = `${this.state.weight} lbs`;
-      const { height, weight, age, gender, dob, zip } = details;
-      const mutation = `
+
+    if (this.isInvalidInput()) {
+      return alert('please enter valid input')
+    }
+
+    let details = this.state
+
+    details.height = `${heightFt}/${heightIn}`
```

Comment:
Try to avoid object mutation when the object is not created and only used within the code block it's created, otherwise it will be very difficult to track where the object mutation comes from when you have several functions across many files that could potentially change the same object!